### ISSUE TITLE
Training issues on custom dataset - Bounding boxes too large & labels not appearing in val_batchX_labels.jpg

### ISSUE THREAD

**Issue by roncamposj** [ultralytics/ultralytics#5877]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I am attempting to finetune a yolov8n model as a hands detector on the [EgoHands Dataset](https://vision.soic.indiana.edu/projects/egohands/) and have manually prepared the dataset for training, following the instructions from https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/#13-prepare-dataset-for-yolov5 . The bounding boxes in the dataset are given as `[x y bbox_width bbox_height] `, with `(x y)` being the coordinates to the top-left corner of the bounding box. I've seen instructions online that claim that:

```
x_center = x + (bbox_width / 2)
y_center = y + (bbox_height / 2)
```

But I believe this is incorrect when the initial coordinates are in the top-left, so instead I am doing:

```
x_center = (x + bbox_width) / 2
y_center = (y + bbox_height) / 2
```

I made a quick script that confirms to me that this approach is right. The red dots are when using my approach while the green are when sticking to the convention.

![centers](https://github.com/ultralytics/ultralytics/assets/42298486/965c61b9-384b-4b8a-97d6-93b18db98777)

I then normalized the datapoints by doing:

```
x_center =  x_center / image_width
y_center = y_center / image_height
bbox_width = bbox_width / image_width
bbox_height = bbox_height / image_height
```
and saved normalized points in a .txt file with the same name as the .jpg image, with the format:

`class x_center y_center width height
`

The images were saved under a folder called `images`, and the bounding boxes in a folder called `labels`.

So my issue is that during and after training (which I do for around 100-150 epochs), the size of my bounding boxes are very large. It does appear as though they are centered on the correct objects, but the size does not appear correct to me, and I believe these are causing issues when attempting to do zero-shot detection on another dataset.

![train_batch1](https://github.com/ultralytics/ultralytics/assets/42298486/b4547faf-15aa-4e50-8b07-c3f4015036bc)

Additionally, I noticed that the labels are rarely appearing in  my val_batchX_labels.jpg files:
![val_batch0_labels](https://github.com/ultralytics/ultralytics/assets/42298486/3b772b15-5649-439a-aa0f-614cae1d9805)

And it's not like the .txt files are missing. For example, this is the corresponding .txt file for jenga_office_b_s_img_0.jpg (
the second image in the first row)
```
0 0.196484 0.909722 0.282031 0.994444
0 0.642969 0.935417 0.712500 0.994444
0 0.556641 0.405556 0.614844 0.498611
0 0.311719 0.427778 0.350000 0.531944
```
I have tweaked around with different hyperparameters such as lowering the learning rate, type of augmentations, and various optimizers, but it seems to me like this is potentially a scaling issue with the bounding boxes. I am confused to why this is happening and would be very grateful for little guidance on how to fix this issue.

### Additional

egohands.yaml
```
# Train/val/test sets as 1) dir: path/to/imgs, 2) file: path/to/imgs.txt, or 3) list: [path/to/imgs1, path/to/imgs2, ..]

path: /home/ron/hands_detection/utils/datasets/egohands  # dataset root dir
train: train/images/  # train images (relative to 'path') 128 images
val: train/images/ # val images (relative to 'path') 128 images
# test:  # test images (optional)

names:
  0: hands
```
I know my train and val point to the same path. The dataset is around 4800 images total, so I want to use as much as I can to train the model.

args.yaml
```
task: detect
mode: train
model: yolov8n.pt
data: /home/ron/hands_detection/egohands.yaml
epochs: 150
patience: 50
batch: 10
imgsz: 640
save: true
save_period: 10
cache: false
device: null
workers: 4
project: null
name: train2
exist_ok: false
pretrained: true
optimizer: SGD
verbose: true
seed: 0
deterministic: true
single_cls: false
rect: false
cos_lr: false
close_mosaic: 10
resume: false
amp: true
fraction: 1.0
profile: false
freeze: null
overlap_mask: true
mask_ratio: 4
dropout: 0.0
val: true
split: val
save_json: false
save_hybrid: false
conf: null
iou: 0.7
max_det: 300
half: false
dnn: false
plots: true
source: null
show: false
save_txt: false
save_conf: false
save_crop: false
show_labels: true
show_conf: true
vid_stride: 1
stream_buffer: false
line_width: null
visualize: false
augment: true
agnostic_nms: false
classes: null
retina_masks: false
boxes: true
format: torchscript
keras: false
optimize: false
int8: false
dynamic: false
simplify: false
opset: null
workspace: 4
nms: false
lr0: 0.005
lrf: 0.01
momentum: 0.937
weight_decay: 0.0005
warmup_epochs: 3.0
warmup_momentum: 0.8
warmup_bias_lr: 0.1
box: 7.5
cls: 0.5
dfl: 1.5
pose: 12.0
kobj: 1.0
label_smoothing: 0.0
nbs: 64
hsv_h: 0.015
hsv_s: 0.7
hsv_v: 0.4
degrees: 0.0
translate: 0.1
scale: 0.5
shear: 0.0
perspective: 0.0
flipud: 0.0
fliplr: 0.5
mosaic: 1.0
mixup: 0.0
copy_paste: 0.0
cfg: null
tracker: botsort.yaml
save_dir: runs/detect/train2

```
![results](https://github.com/ultralytics/ultralytics/assets/42298486/d547ebbe-cfbc-4e13-b298-bbe811da0e5e)

**Comment by roncamposj** [comment]
I figured out the issue with the bounding boxes. I was passing the wrong `width` and `height` values to my .txt files. Feel free to mark this as resolved.

**Comment by glenn-jocher** [comment]
@roncamposj that's great to hear! I'm glad you were able to figure out the issue with the bounding boxes. Thank you for letting us know. Remember, if you have any more questions in the future, or if you encounter any other issues, don't hesitate to reach out. We're here to help. Marking this as resolved. Happy coding!

**Comment by vTuanpham** [comment]
I'm having a similar issue, the train_batch and the validation set as long as the model final cpkt is messed up even though when i try to plot the label separately, all seem to checkout

![Image](https://github.com/user-attachments/assets/0f651113-cd13-4780-ab20-79658690bf86)

**Comment by glenn-jocher** [comment]
@vTuanpham Thanks for sharing your issue. This appears to be a common problem related to coordinate conversion or normalization. Since you mentioned the labels check out when plotted separately, the issue might be in how they're being interpreted during training.

Some things to check:
1. Ensure your bounding box format follows YOLO's normalized xywh format (x_center/img_width, y_center/img_height, width/img_width, height/img_height)
2. Verify your coordinate calculation for center points is correct (as the original poster discovered)
3. Check that image dimensions used for normalization match the actual dimensions loaded during training

You could also try running the validation with the `--rect` flag disabled if you're using it, as this can sometimes affect visualization. If you need to validate your labels are correct, you can use Ultralytics' utility functions to convert and visualize them:

```python
from ultralytics.data.utils import autosplit
from ultralytics.utils.ops import segments2boxes
```

Let me know if you need more specific guidance on any part of your setup.

**Comment by vTuanpham** [comment]
Don't give me such generic gpt response, this is such a simple issue and the dataset is actually exported from roboflow, which is also one of your own product

**Comment by glenn-jocher** [comment]
I apologize for the generic response, @vTuanpham. You're right to expect more specific help.

First, to clarify - Roboflow and Ultralytics are separate companies, though we do collaborate on some projects.

For datasets exported from Roboflow specifically, the visualization issue you're seeing despite correct labels often comes from mismatches between the annotation format and how the model interprets them. Since you mentioned your separate visualization looks correct, let's focus on fixing the training display.

Could you check if you selected "YOLO v8" as the exact export format in Roboflow? Sometimes selecting an older YOLO version can cause these visualization issues. Also, try examining a few specific label files directly to verify their exact format matches what's expected by YOLO11.

If you can share a sample label file and the exact Roboflow export settings you used, I can provide more targeted assistance.
